File: lint-arwaky/archive/infrastructure/arch_compliance_adapter.py
"""Infrastructure adapter that wraps the architectural compliance capability."""

from __future__ import annotations

from ..taxonomy import (
    AdapterName,
    ComplianceStatus,
    FilePath,
    LintResultList,
    AdapterError,
    ScanError,
)
from ..contract import ILinterAdapterPort, IArchCompliancePort


class ArchComplianceAdapter(ILinterAdapterPort):
    """Adapter that bridges architectural compliance to the linter pipeline.
    
    Implements ILinterAdapterPort by delegating to an IArchCompliancePort.
    """

    def __init__(self, coordinator: IArchCompliancePort):
        self.coordinator = coordinator

    def name(self) -> AdapterName:
        return AdapterName(value="architecture")

    async def scan(self, path: FilePath) -> LintResultList | ScanError | AdapterError:
        """Performs architectural scan using the compliance coordinator."""
        try:
            return await self.coordinator.scan(path)
        except Exception as e:
            return AdapterError(adapter_name="architecture", message=f"Architecture scan failed: {str(e)}")

    async def apply_fix(self, path: FilePath) -> ComplianceStatus | AdapterError:
        """Execute architectural fix."""
        try:
            return await self.coordinator.apply_fix(path)
        except Exception as e:
            return AdapterError(adapter_name="architecture", message=f"Architecture fix failed: {str(e)}")

File: lint-arwaky/src-rust/infrastructure/arch_compliance_adapter.rs
/// arch_compliance_adapter — Infrastructure adapter that wraps architectural compliance checking.
use crate::contract::{IArchCompliancePort, ILinterAdapterPort, LinterError};
use crate::taxonomy::{AdapterError, AdapterName, ComplianceStatus, ErrorMessage, FilePath, LintResultList, ScanError};
use std::sync::Arc;

pub struct ArchComplianceAdapter {
    coordinator: Arc<dyn IArchCompliancePort>,
}

impl ArchComplianceAdapter {
    pub fn new(coordinator: Arc<dyn IArchCompliancePort>) -> Self {
        Self { coordinator }
    }
}

#[async_trait::async_trait]
impl ILinterAdapterPort for ArchComplianceAdapter {
    fn name(&self) -> AdapterName {
        AdapterName::new("architecture")
    }

    async fn scan(&self, path: &FilePath) -> Result<LintResultList, LinterError> {
        Ok(self.coordinator.scan(path).await)
    }

    async fn apply_fix(&self, path: &FilePath) -> Result<ComplianceStatus, LinterError> {
        Ok(self.coordinator.apply_fix(path).await)
    }
}
